File: lab2/lab2.py
def invert_and_sum():
    arr = [0, 254, 126, 126, 126, 126, 126, 126, 126, 254]
    sum_val = sum([val if not (val & (1<<7)) else (~val & 0xFF) for val in arr])
    print(sum_val)

if __name__ == '__main__':
    import dis
    print(dis.dis(invert_and_sum))

File: lab2/lab2.cpp
#include <stdio.h>

int main(){
    int array[10] = {0, 254, 126, 126, 126, 126, 126, 126, 126, 254};
    int sum_value = 0;
    for (int i = 0; i < 10; i++){
        if (array[i] & 128){
            sum_value += (~array[i] & 0xFF);
        }
        else sum_value += array[i];
    }
    printf("Sum: %u ", sum_value);
    return 0;

}